docs: fix accidental commas and add extre note to GRACE PERIOD syntax (#7706)
confluentinc/ksql: docs/developer-guide/ksqldb-reference/select-push-query.md

        INNER JOIN shipments s WITHIN 2 HOURS ON s.id = o.id;
```

The GRACE PERIOD, part of the WITHIN clause, allows the join to process out-of-order records for up
to the specified grace period. Events that arrive after the grace period has passed are dropped
as _late_ record and not joined.

> The default grace period, i.e., if no explicitly specified, is 24 hours. This could cause a huge

Reviewer: ```suggestion
The default grace period, if you don't specify it explicitly, is 24 hours. This could cause a huge

```

amount of disk usage on high-throughput streams. Setting a specific GRACE PERIOD is recommended to
reduce high disk usage.

!!! note

>     If you specify a GRACE PERIOD for left/outer joins, the grace period define when left/outer

Reviewer: ```suggestion
    If you specify a GRACE PERIOD for left/outer joins, the grace period defines when the left/outer

```

>     join result will be emitted. If you don't specify a GRACE PERIOD for left/outer joins,

Reviewer: ```suggestion
    join result is emitted. If you don't specify a GRACE PERIOD for left/outer joins,

```

>     left/outer join results are emitted eagerly, what may lead to "spurious" result records. Thus,

Reviewer: ```suggestion
    left/outer join results are emitted eagerly, which may cause "spurious" result records, so

```

>     it's highly recommended to specify a GRACE PERIOD. For this reason, we plan to make

Reviewer: ```suggestion
    we recommended that you specify a GRACE PERIOD.

```

>     GRACE PERIOD a mandatory clause in the future.

Reviewer: ```suggestion

```

Reviewer: We try not to comment on future features in docs.

Author: ok

        INNER JOIN shipments s WITHIN 2 HOURS ON s.id = o.id;
```

The GRACE PERIOD, part of the WITHIN clause, allows the join to process out-of-order records for up
to the specified grace period. Events that arrive after the grace period has passed are dropped

> as _late_ record and not joined.

Reviewer 2: `record[s]`

> If you don't specify it explicitly, the default grace period is 24 hours. This could cause a huge

Reviewer 2: `it` -> `a grace period`
